src/components/Notifications.tsx

```diff
@@ -0,0 +1,55 @@
+const Notifications = Windows.UI.Notifications;
+const ToastTemplateType = Notifications.ToastTemplateType;
+const ToastNotificationManager = Notifications.ToastNotificationManager;
+const ToastNotification = Notifications.ToastNotification;
+
+export function showNotification(notification) {
+  var type = ToastTemplateType.toastText01;
+
+  var obj = {};
+  if (typeof notification === 'string') {
+    obj.text = notification;
+  } else {
+    obj = notification;
+  }
+
+  if (obj.template != undefined) {
+    type = obj.template;
+  }
+
+  var xml = ToastNotificationManager.getTemplateContent(type);
+  for (var tagName in obj) {
+    var xmlElements = xml.getElementsByTagName(tagName);
+    var value = obj[tagName];
+    if (typeof value === 'string') {
+      fillXmlElements(xml, xmlElements, [value]);
+    } else if (Array.isArray(value)) {
+      fillXmlElements(xml, xmlElements, value);
+    } else if (typeof value === 'object') {
+      fillXmlElements(xml, xmlElements, [value]);
+    }
```

Comment:
Looks like case 1 and 3 are the same. Is this still correct?
```suggestion
    if (Array.isArray(value)) {
      fillXmlElements(xml, xmlElements, value);
    } else {
      fillXmlElements(xml, xmlElements, [value]);
    }
```
Or do you get types other than string/object in there? (like numeric literals or something)

Reply:
If your goal is "if it's an array, leave it; otherwise, place in an array" there might be other options. Like:
```
[...value]
```
Does that do the right thing? It definitely spreads a source array into a new array. But for a single item... I think still the same thing? Hmmm... except strings are iterable so that'll break apart the string. Anyway, might be some syntactic magic here.